offroad/js/components/Settings/Settings.js

```diff
@@ -588,6 +614,105 @@ class Settings extends Component {
+                        onChangeText={ (text) => this.setState({ githubUsername: text, authKeysUpdateState: null })}
+                        value={ githubUsername }
+                        ref={ ref => this.githubInput = ref }
+                        underlineColorAndroid='transparent'
+                    />
+                </View>
+                <View>
+                    <X.Button
+                        size='tiny'
+                        color='settingsDefault'
+                        isDisabled={ !githubUsernameIsValid }
+                        onPress={ this.writeSshKeys }
+                        style={ Styles.githubUsernameSaveButton }>
+                        <X.Text color='white' size='small' style={ Styles.githubUsernameInputSave }>Save</X.Text>
+                    </X.Button>
+                    { authKeysUpdateState !== null &&
+                        <View style={ Styles.githubUsernameInputStatus }>
+                            { authKeysUpdateState === 'inflight' &&
+                                <ActivityIndicator
+                                    color='white'
+                                    refreshing={ true }
+                                    size={ 37 }
+                                    style={ Styles.connectingIndicator } />
+                            }
+                            { authKeysUpdateState === 'failed' &&
+                                <X.Text color='white' size='tiny'>Save failed. Ensure that your username is correct and you are connected to the internet.</X.Text>
+                            }
+                        </View>
+                    }
+                    <View style={ Styles.githubSshKeyClearContainer }>
+                        <X.Button
+                            size='tiny'
+                            color='settingsDefault'
+                            onPress={ this.clearSshKeys }
+                            style={ Styles.githubUsernameSaveButton }>
+                            <X.Text color='white' size='small' style={ Styles.githubUsernameInputSave }>Remove all</X.Text>
+                        </X.Button>
+                    </View>
+                </View>
+            </View>
+        );
+    }
+
+    toggleExpandGithubInput() {
+        this.setState({ authKeysUpdateState: null });
+        this.handleExpanded('ssh_keys');
+    }
+
+    clearSshKeys() {
+        ChffrPlus.deleteParam(Params.KEY_GITHUB_SSH_KEYS);
+        Alert.alert('Reboot', `Reboot to finalize removal of GitHub SSH keys.`, [
+            { text: 'Later', onPress: () => {}, style: 'cancel' },
+            { text: 'Reboot Now', onPress: () => ChffrPlus.reboot() },
+        ]);
+    }
+
+    async writeSshKeys() {
+        let { githubUsername } = this.state;
+
+        try {
+            this.setState({ authKeysUpdateState: 'inflight' })
+            const resp = await fetch(`https://github.com/${githubUsername}.keys`);
```

Comment:
Not sure what our threat model here, but it feels a bit weird to not do any sanity checking on the user supplied parameter.

For more info about valid usernames: https://github.com/shinnn/github-username-regex

> According to the form validation messages on Join Github page,
> * Github username may only contain alphanumeric characters or hyphens.
> * Github username cannot have multiple consecutive hyphens.
> * Github username cannot begin or end with a hyphen.
> * Maximum is 39 characters.

Reply:
We sanitize the username with `githubUsername.match(/[a-zA-Z0-9-]+/) !== null;` - cannot enter writeSshKeys without this passing.